Google.PowerShell/Container/GkeClusterCmdlets.cs

```diff
@@ -580,4 +581,381 @@ protected override void ProcessRecord()
+            RuntimeDefinedParameterDictionary dynamicParamDict)
+        {
+            // Gets all the valid machine types of this zone and project combination.
+            string[] machineTypes = GetMachineTypes(Project, Zone);
+            RuntimeDefinedParameter machineTypeParam = GenerateRuntimeParameter(
+                parameterName: "MachineType",
+                helpMessage: "The Google Compute Engine machine type to use for node in this cluster.",
+                parameterSetName: ParameterSetNames.ByValues,
+                validSet: machineTypes);
+            dynamicParamDict.Add("MachineType", machineTypeParam);
+
+            // Gets all the valid image types of this zone and project combination.
+            string[] imageTypes = GetImageTypes(Project, Zone);
+            RuntimeDefinedParameter imageTypeParam = GenerateRuntimeParameter(
+                parameterName: "ImageType",
+                helpMessage: "The image type to use for node in this cluster.",
+                parameterSetName: ParameterSetNames.ByValues,
+                validSet: imageTypes);
+            dynamicParamDict.Add("ImageType", imageTypeParam);
+        }
+
+        protected override void ProcessRecord()
+        {
+            try
+            {
+                CreateClusterRequest requestBody = new CreateClusterRequest() { Cluster = BuildCluster() };
+                ProjectsResource.ZonesResource.ClustersResource.CreateRequest request =
+                    Service.Projects.Zones.Clusters.Create(requestBody, Project, Zone);
+                Operation createOperation = request.Execute();
+                Cluster createdCluster = WaitForClusterCreation(createOperation);
+                WriteObject(createdCluster);
+            }
+            catch (GoogleApiException apiEx) when (apiEx.HttpStatusCode == HttpStatusCode.Conflict)
+            {
+                WriteResourceExistsError(
+                    exceptionMessage: $"Cluster '{ClusterName}' already exists in zone '{Zone}' of project '{Project}'",
+                    errorId: "ClusterAlreadyExists",
+                    targetObject: ClusterName);
+            }
+        }
+
+        /// <summary>
+        /// Wait for the cluster creation operation to complete.
+        /// Use write progress to display the progress in the meantime.
+        /// </summary>
+        public Cluster WaitForClusterCreation(Operation operation)
+        {
+            int activityId = (new Random()).Next();
+            string activity = $"Creating cluster '{ClusterName}' in zone '{Zone}' of project '{Project}'.";
+            string status = "Creating cluster";
+            int percentage = 0;
+
+            while (operation.Status != "DONE")
+            {
+                Thread.Sleep(200);
+                var progressRecord = new ProgressRecord(activityId, activity, status);
+                percentage = (percentage + 1) % 100;
```

Comment:
Yes, it is because we don't know in advance how long it will take to create the cluster.